src/cliffordsynthesis/encoding/SingleGateEncoder.cpp

```diff
@@ -119,6 +155,33 @@ LogicTerm SingleGateEncoder::createNoChange(
   return changes;
 }
 
+LogicTerm SingleGateEncoder::createNoChangeOnQubit(const std::size_t pos,
+                                                   const std::size_t q) {
+  auto noChange = LogicTerm(true);
+  noChange      = noChange && (tvars->x[pos + 1][q] == tvars->x[pos][q]);
+  noChange      = noChange && (tvars->z[pos + 1][q] == tvars->z[pos][q]);
+  return noChange;
+}
+
+LogicTerm SingleGateEncoder::createNoGateOnQubit(const std::size_t pos,
+                                                 const std::size_t q) {
+  const auto& singleQubitGates = vars.gS[pos];
+  auto        noGate           = LogicTerm(true);
+  for (std::size_t i = 1; i < SINGLE_QUBIT_GATES.size(); ++i) {
+    noGate = noGate && !singleQubitGates[i][q];
+  }
+  const auto& twoQubitGates = vars.gC[pos];
+  for (std::size_t i = 0; i < N; ++i) {
+    if (i == q) {
+      continue;
+    }
+    noGate = noGate && !twoQubitGates[i][q];
+    noGate = noGate && !twoQubitGates[q][i];
+  }
```

Comment:
Probably stupid, but for larger qubit numbers, this is quite a long chain of ANDs. Would something like a commander encoding (i.e. some helper variables) be helpful here?
Or would you say that long chains of ANDs aren't really a problem?

Reply:
In this particular case I would argue that it probably doesn't hurt too much. In my experience a commander encoding is useful for formulas that grow quadratically or worse. I'm not sure if grouping here would gain us much. 

I would say that this is something to benchmark for the future but not a pressing optimization.